### PR TITLE
added context example to README

### PR DESCRIPTION
**Pull request by nfcampos**
@ryanbrainard I finally got around to writing an example for context, what do you think, is this what you had in mind?

(related to https://github.com/heroku/react-refetch/pull/86)

I've also made two other changes in this PR (I wans't sure if I should open separate PRs for such small things):
- added the `curly` eslint rule as we discussed in a previous PR
- fixed `npm run test:cov` which wasn't working

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -341,6 +341,83 @@ userFetch.meta.response.headers.get('FOO')
 
 Do not attempt to read bodies directly from `meta.request` or `meta.response`. They are provided for metadata purposes only.
 
+## Using context to define requests
+
+You can also use `context` (in addition to props) to help define the requests your components need.
+It's available as the second argument to the function you pass to `connect`.
+Don't forget to define `contextTypes` on the component you're wrapping for context to be available.
+
+Some use case examples are, eg. setting a urlPrefix in a root component that all others use:
```
Comment (ryanbrainard) [inline README.md]: s/urlPrefix/URL prefix/

File: README.md [hunk 2]
```diff
@@ -341,6 +341,83 @@ userFetch.meta.response.headers.get('FOO')
...
+    userFetch: `/user/${props.params.userId}`,
+    likesFetch: {
+      url: `/users/${props.userId}/likes`,
+      refreshInterval: context.refreshInterval,
+    }
+  }
+})(Profile)
+```
+
+This is a complete example of everything you need to use context with react-refetch:
```
Comment (ryanbrainard) [inline README.md]: s/react-refetch/React Refetch/

**Comment by ryanbrainard** [comment]
Looks good other than two simple little style nitpicks. 

Once #76 is pulled in, we might want to add some explanation explaining why you would set defaults in `connect.default` vs `context`, but we can do that later.

**Comment by nfcampos** [comment]
I can add that note now if you want. The reason for doing it via context is basically so that it can change dynamically, right?

**Comment by ryanbrainard** [comment]
> The reason for doing it via context is basically so that it can change dynamically, right?

Dynamically and/or more locally (depending on how global your defaults are). 

> I can add that note now if you want.

Let's hold off. I'm going to pull this in now, and push a release so that #76 will have its own beta.

**Comment by nfcampos** [comment]
one other reason i can think of to use context:
an example like the default refreshTimeout I added to the readme in which you want to have a default for some mappings is impossible to do with `.defaults()`

**Comment by ryanbrainard** [comment]
Yep, for sure. I like this use of context for defaults as you've shown.

Released: https://github.com/heroku/react-refetch/releases/tag/v0.8.0

**Comment by loris** [comment]
@ryanbrainard @nfcampos Thanks for the example, I was wondering if there is a more DRYer way to make react-refetch access values in the context (in my case, authentication data such as API tokens) without explicitly doing this for every components using react-refetch (I have hundreds of them in a large codebase):

```
static contextTypes = {
  apiToken: PropTypes.string.isRequired,
}
```

I was thinking of wrapping my component with a "child-as-a-function" component (see example here: https://github.com/heroku/react-refetch/issues/140#issuecomment-245650583) instead of the HOC, what do you think?

**Pull request merged**